src/iam/iam-base.ts

```diff
@@ -262,31 +273,58 @@ export class IAMBase {
         gasLimit: hexlify(4900000),
         gasPrice: hexlify(0.1)
       };
+
+      // Don't show QR code if using KeyManager because we send user directly to app instead
+      const showQRCode = !(walletProvider === WalletProvider.EwKeyManager);
+
       this._walletConnectProvider = new WalletConnectProvider({
         rpc: {
           [this._connectionOptions.chainId]: this._connectionOptions.rpcUrl
         },
         infuraId: this._connectionOptions.infuraId,
-        bridge: this._connectionOptions.bridgeUrl
+        bridge: this._connectionOptions.bridgeUrl,
+        qrcode: showQRCode
       });
+
+      if (walletProvider === WalletProvider.EwKeyManager) {
+        this._walletConnectProvider.wc.on("display_uri", (err, payload) => {
+          // uri is expected to be WalletConnect Standard URI https://eips.ethereum.org/EIPS/eip-1328
+          const wcUri = payload.params[0];
+          console.log(wcUri);
```

Comment:
If not needed please remove console.log 😄

Reply:
Thanks, good catch, I've removed it.